docs/architecture.md

```diff
@@ -0,0 +1,79 @@
+# Architecture
+
+If you haven't already, please see the [guidelines document](./guidelines.md) first.
+
+## Crates and modules
+
+The codebase would be simpler without crates. I'd like to make the following arguments for
+modules and against crates:
+
+- Dead code analysis
+- Having to explicitly encode dependencies between crates
+- Workspace management is more complex than managing a single binary
+
+Instead I propose we use command line arguments to run specific subsystems:
+
+```
+$ binary_name command_name --flags
+```
+
+And collapse all of our crates into a single crate with modules.
+
```

Comment:
Another argument in favour of crates is that it allows us to export some logic for others to integrate (conceivably, an orderbook API Rust client for example, with shared `model` types).